### PR TITLE
Add azure-storage-cli to the stemcells

### PR DESCRIPTION
**Pull request by ten4o**
Add blobstore client for Azure aka [azure-storage-cli](https://github.com/cloudfoundry/bosh-azure-storage-cli) binary to the stemcells.

Related PRs
 - bosh-stemcells-ci [Add azure-storage-cli to the stemcells](https://github.com/cloudfoundry/bosh-stemcells-ci/pull/14)

### CODE REVIEW

**Review by Malsourie — APPROVED** [review]
For integrating `azure-storage-cli` into stemcell-builder, LGTM.
In the future, the three assets files `.sha256sum`, `.url` and `.version` should be maintained by [bosh-stemcells-ci](https://github.com/cloudfoundry/bosh-stemcells-ci).

File: bosh-stemcell/spec/stemcells/go_agent_spec.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 describe 'Stemcell with Go Agent', stemcell_image: true do
   describe 'installed by bosh_go_agent' do
-    %w(bosh-agent bosh-agent-rc bosh-blobstore-dav bosh-blobstore-gcs bosh-blobstore-s3).each do |binary|
+    %w(bosh-agent bosh-agent-rc bosh-blobstore-dav bosh-blobstore-gcs bosh-blobstore-s3 bosh-blobstore-az).each do |binary|
```
Comment (Malsourie) [inline bosh-stemcell/spec/stemcells/go_agent_spec.rb]: Please adjust the package name to `bosh-blobstore-azure-storage` corresponding to the definition in azure-storage-cli repo.

File: stemcell_builder/stages/blobstore_clis/config.sh [hunk 2]
```diff
@@ -11,7 +11,7 @@ persist_value downloaded_cli_dir
 rm -rf "${downloaded_cli_dir}"
 mkdir -p "${downloaded_cli_dir}"
 
-blobstore_clis=(bosh-blobstore-dav bosh-blobstore-gcs bosh-blobstore-s3)
+blobstore_clis=(bosh-blobstore-dav bosh-blobstore-gcs bosh-blobstore-s3 bosh-blobstore-az)
```
Comment (Malsourie) [inline stemcell_builder/stages/blobstore_clis/config.sh]: Please adjust the package name to `bosh-blobstore-azure-storage` corresponding to the definition in azure-storage-cli repo.

**Review by Malsourie — CHANGES_REQUESTED** [review]

**Comment by Malsourie** [comment]
Please also adapt the filename of three assets files `.sha256sum`, `.url` and `.version` to bosh-blobstore-azure-storage

**Comment by rkoster** [comment]
This PR should be coordinated with the agent changes.

File: bosh-stemcell/spec/stemcells/go_agent_spec.rb (hunk 1 above)
Comment (ten4o) [reply bosh-stemcell/spec/stemcells/go_agent_spec.rb]: Done.

File: stemcell_builder/stages/blobstore_clis/config.sh (hunk 2 above)
Comment (ten4o) [reply stemcell_builder/stages/blobstore_clis/config.sh]: Done.

**Comment by ramonskie** [comment]
and it should also coordinated with the changes to the stemcell ci builder
see https://github.com/cloudfoundry/bosh-stemcells-ci/blob/master/pipelines/ubuntu-jammy/pipeline.yml#L1399
and https://github.com/cloudfoundry/bosh-stemcells-ci/blob/master/pipelines/ubuntu-jammy/stemcells.yml#L31
for the changes needed

**Review by Malsourie — APPROVED** [review]

**Comment by mvach** [comment]
lgtm

**Review by mvach — APPROVED** [review]

File: stemcell_builder/stages/blobstore_clis/assets/bosh-blobstore-azure-storage.version [hunk 3]
```diff
@@ -0,0 +1 @@
+0.0.7
```
Comment (mvach) [inline stemcell_builder/stages/blobstore_clis/assets/bosh-blobstore-azure-storage.version:1]: Just noticed That there is a 0.0.8 already

Comment (ramonskie) [reply stemcell_builder/stages/blobstore_clis/assets/bosh-blobstore-azure-storage.version:1]: this should auto update later on when the https://github.com/cloudfoundry/bosh-stemcells-ci/pull/14 pr is merged

**Review by rkoster — APPROVED** [review]

**Review by ramonskie — APPROVED** [review]
lgtm

**Comment by rkoster** [comment]
Thanks! @ten4o

**Pull request merged**